src/software/multithreading/ordered_threaded_observer.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <boost/bind.hpp>
+#include <thread>
+
+#include "software/multithreading/observer.h"
+
+/**
+ * The general usage of this class should be to extend it, then override
+ * `onValueReceived` with whatever custom functionality should occur when a new value
+ * is received.
+ *
+ * @tparam T The type of object this class is observing
+ */
+template <typename T>
+class OrderedThreadedObserver : public Observer<T>
```

Comment:
Sorry, not quite clear on what the difference is between this class and the normal `ThreadedObserver`?

Reply:
buffer size + `popLeastRecentlyAddedValue`

Reply:
Mentioned in my above comment, but since `popLeastRecentlyAddedValue` is implemented in the Observer class and `buffer_size` is already a part of the interface, I don't think we need a separate class.